server/app/core/usecases.py

```diff
@@ -141,64 +153,410 @@ async def save(
+        resource_path = resource_path.strip("/")
+        related_folder = related_folder.strip("/")
+
+        # Compare using pathlib, which handles "../" etc cleanly
+        resource_parts = PurePosixPath(resource_path).parts
+        folder_parts = PurePosixPath(related_folder).parts
+
+        # Must start with the folder parts
+        return resource_parts[: len(folder_parts)] == folder_parts
+
+    def _update_distribution_in_place(
+        self,
+        graph,
+        dist_node,
+        connector_data,
+        existing_data,
+        dataset_jsonld,
+    ):
+        """
+        Ontology-agnostic update of a DCAT Distribution.
+
+        Rules:
+        - If connector provides a non-null value => override.
+        - If connector provides null and dataset has value => keep dataset.
+        - If both connector and dataset have null => explicitly set 'null' (typed).
+        - Keep RDF types (URIRefs for URLs, typed literals otherwise).
+        - Update nested checksum node correctly (no duplicate checksumValue).
+        """
+
+        def add_or_replace_literal(
+            subject, predicate, value, dtype=XSD.string, is_uri=False
+        ):
+            """
+            Helper: replace triple or insert new, respecting RDF typing.
+            """
+            # Remove any existing values first
+            graph.remove((subject, predicate, None))
+
+            # 1 Both null => explicit "null"
+            if value is None or str(value).lower() in ("none", ""):
+                graph.add((subject, predicate, Literal("null", datatype=dtype)))
+
+                return
+
+            # 2 URIs (for URLs)
+            if is_uri:
+                graph.add((subject, predicate, URIRef(value)))
+
+            else:
+                graph.add((subject, predicate, Literal(value, datatype=dtype)))
+
+        # DCAT-AP 3.0.0 field mapping (minimal set)
+        field_map = {
+            "access_url": (DCAT.accessURL, XSD.anyURI, True),
+            "download_url": (DCAT.downloadURL, XSD.anyURI, True),
+            "media_type": (DCAT.mediaType, XSD.string, False),
+            "package_format": (DCAT.packageFormat, XSD.string, False),
+            "format": (DCTERMS.format, XSD.string, False),
+            "byte_size": (DCAT.byteSize, XSD.long, False),
+            "issued": (DCTERMS.issued, XSD.dateTime, False),
+            "modified": (DCTERMS.modified, XSD.dateTime, False),
+            "title": (DCTERMS.title, XSD.string, False),
+            "description": (DCTERMS.description, XSD.string, False),
+            "license": (DCTERMS.license, XSD.string, False),
+            "rights": (DCTERMS.rights, XSD.string, False),
+            "conforms_to": (DCTERMS.conformsTo, XSD.anyURI, True),
+            "access_rights": (DCTERMS.accessRights, XSD.string, False),
+            "has_policy": (DCATAP.hasPolicy, XSD.anyURI, True),
+            "compress_format": (DCAT.compressFormat, XSD.anyURI, True),
+        }
+
```

Comment:
@renuka-gollu If we moved this mapping in configuration file 
it will me more agnotstic as we load this mapping and we use it in the code directly 
someting like this 
config.json
```
{
  "access_url": ["http://www.w3.org/ns/dcat#accessURL", "anyURI", true],
  "media_type": ["http://www.w3.org/ns/dcat#mediaType", "string", false],
  "byte_size": ["http://www.w3.org/ns/dcat#byteSize", "long", false]
}

```

then you Load it in the code and continue the logic 
```
with open("config.json") as f:
    field_map = json.load(f)
```